### PR TITLE
SPM: update DML SQL Bind and baseline capture description

### PR DESCRIPTION
**Pull request by CharLotteiu**
### What is changed, added or deleted? (Required)

Update SPM documentation for DML SQL Bind and baseline capture

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/5740
- Other reference link(s):  https://github.com/pingcap/tidb/pulls?q=is%3Apr+is%3Aclosed+label%3Aepic%2Fsql-plan-management

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-plan-management.md [hunk 1]
```diff
@@ -17,10 +17,73 @@ An SQL binding is the basis of SPM. The [Optimizer Hints](/optimizer-hints.md) d
...
+CREATE [GLOBAL | SESSION] BINDING FOR BindableStmt USING BindableStmt
 ```
 
-This statement binds SQL execution plans at the GLOBAL or SESSION level. The default scope is SESSION. The bound SQL statement is parameterized and stored in the system table. When a SQL query is processed, as long as the parameterized SQL statement and a bound one in the system table are consistent and the system variable `tidb_use_plan_baselines` is set to `on` (the default value is `on`), the corresponding optimizer hint is available. If multiple execution plans are available, the optimizer chooses to bind the plan with the least cost.
+This statement binds SQL execution plans at the GLOBAL or SESSION level. Currently, supported bindable SQL statements (BindableStmt) in TiDB include `SELECT`, `DELETE`, `UPDATE`, and `INSERT` / `REPLACE` with `SELECT` subqueries.
+
+Specifically, two types of these statements cannot be bound to execution plans due to syntax conflicts. See the following examples:
+
+```sql
+-- Type one: Statements that use the `join` keyword to get the Cartesian product but not specify the associated columns with the `using` keyword.
```
Comment (eurekaka) [inline sql-plan-management.md:28]:
```suggestion
-- Type one: Statements that use the `join` keyword but not specify the associated columns with the `using` keyword, to get the Cartesian product.
```

File: sql-plan-management.md [hunk 2]
```diff
@@ -17,10 +17,73 @@ An SQL binding is the basis of SPM. The [Optimizer Hints](/optimizer-hints.md) d
...
+using
+    insert into t1 select /*+ use_index(@sel_1 t2, a) */ * from t2 where a > 1 and b = 1;
+-- The hint cannot take effect in the following statement.
+create global binding for
+    insert into t1 select * from t2 where a > 1 and b = 1
+using
+    insert /*+ use_index(@sel_1 t2, a) */ into t1 select * from t2 where a > 1 and b = 1;
+```
+
+If you do not specify the scope when creating an execution plan binding, the default scope is SESSION. The TiDB optimizer standardizes bound SQL statements and stores them in the system table. When processing SQL queries, if a standardized statement matches one of the bound SQL statements in the system table and the system variable `tidb_use_plan_baselines` is set to `on` (the default value is `on`), TiDB then uses the corresponding optimizer hint for this statement. If there are multiple matchable execution plans, the optimizer choose the least costly one to bind.
```
Comment (eurekaka) [inline sql-plan-management.md:78]:
```suggestion
If you do not specify the scope when creating an execution plan binding, the default scope is SESSION. The TiDB optimizer normalizes bound SQL statements and stores them in the system table. When processing SQL queries, if a normalized statement matches one of the bound SQL statements in the system table and the system variable `tidb_use_plan_baselines` is set to `on` (the default value is `on`), TiDB then uses the corresponding optimizer hint for this statement. If there are multiple matchable execution plans, the optimizer choose the least costly one to bind.
```

File: sql-plan-management.md [hunk 3]
```diff
@@ -17,10 +17,73 @@ An SQL binding is the basis of SPM. The [Optimizer Hints](/optimizer-hints.md) d
...
+
+`Standardization` is a process that converts a constant in an SQL statement to a variable parameter and explicitly specifies the database for tables referenced in the query, with standardized processing on the spaces and line breaks in the SQL statement. See the following example:
```
Comment (eurekaka) [inline sql-plan-management.md:80]:
```suggestion
`Normalization` is a process that converts a constant in an SQL statement to a variable parameter and explicitly specifies the database for tables referenced in the query, with standardized processing on the spaces and line breaks in the SQL statement. See the following example:
```

File: sql-plan-management.md [hunk 4]
```diff
@@ -17,10 +17,73 @@ An SQL binding is the basis of SPM. The [Optimizer Hints](/optimizer-hints.md) d
...
+
+```sql
+select * from t where a >    1
+-- Standardized:
```
Comment (eurekaka) [inline sql-plan-management.md:84]:
```suggestion
-- Normalized:
```

File: sql-plan-management.md [hunk 5]
```diff
@@ -48,42 +111,36 @@ explain select * from t1, t2 where t1.id = t2.id;
...
-```
-
 Each standardized SQL statement can have only one binding created using `CREATE BINDING` at a time. When multiple bindings are created for the same standardized SQL statement, the last created binding is retained, and all previous bindings (created and evolved) are marked as deleted. But session bindings and global bindings can coexist and are not affected by this logic.
 
 In addition, when you create a binding, TiDB requires that the session is in a database context, which means that a database is specified when the client is connected or `use ${database}` is executed.
 
 > **Note:**
 >
-> The text must be the same before and after parameterization and hint removal for both the original SQL statement and the bound statement, or the binding will fail. Take the following examples:
+> The text must be the same before and after standardization and hint removal for both the original SQL statement and the bound statement, or the binding will fail. Take the following examples:
```
Comment (eurekaka) [inline sql-plan-management.md:120]:
```suggestion
> The text must be the same before and after normalization and hint removal for both the original SQL statement and the bound statement, or the binding will fail. Take the following examples:
```

File: sql-plan-management.md [hunk 6]
```diff
@@ -132,7 +189,15 @@ To enable baseline capturing, set `tidb_capture_plan_baselines` to `on`. The def
 >
 > Because the automatic binding creation function relies on [Statement Summary](/statement-summary-tables.md), make sure to enable Statement Summary before using automatic binding.
 
-After automatic binding creation is enabled, the historical SQL statements in the Statement Summary are traversed every `bind-info-lease` (the default value is `3s`), and baseline capturing is automatically created for SQL statements that appear at least twice.
+After automatic binding creation is enabled, the historical SQL statements in the Statement Summary are traversed every `bind-info-lease` (the default value is `3s`), and baseline capturing is automatically created for SQL statements that appear at least twice. For these SQL statements, TiDB automatically binds the execution plan recorded in Statement Summary.
```
Comment (eurekaka) [inline sql-plan-management.md:192]:
```suggestion
After automatic binding creation is enabled, the historical SQL statements in the Statement Summary are traversed every `bind-info-lease` (the default value is `3s`), and a binding is automatically created for SQL statements that appear at least twice. For these SQL statements, TiDB automatically binds the execution plan recorded in Statement Summary.
```

File: sql-plan-management.md [hunk 7]
```diff
@@ -132,7 +189,15 @@ To enable baseline capturing, set `tidb_capture_plan_baselines` to `on`. The def
...
+
+However, TiDB does not automatically capture bindings from the baseline for the following types of SQL statements:
```
Comment (eurekaka) [inline sql-plan-management.md:194]:
```suggestion
However, TiDB does not automatically capture bindings for the following types of SQL statements:
```

File: sql-plan-management.md [hunk 8]
```diff
@@ -132,7 +189,15 @@ To enable baseline capturing, set `tidb_capture_plan_baselines` to `on`. The def
...
+
+- `EXPLAIN` and `EXPLAIN ANALYZE` statements.
+- SQL statements executed internally in TiDB, such as `SELECT` queries used for automatically loading statistical information.
+- SQL statements that are bound to a manually created execution plan.
+
+For `PREPARE` / `EXECUTE` statements and queries executed with binary protocols, TiDB automatically captures bindings from the baseline for real query statements, not for the `PREPARE` / `EXECUTE` statements.
```
Comment (eurekaka) [inline sql-plan-management.md:200]:
```suggestion
For `PREPARE` / `EXECUTE` statements and queries executed with binary protocols, TiDB automatically captures bindings for the real query statements, not for the `PREPARE` / `EXECUTE` statements.
```

File: sql-plan-management.md (hunk 1 above)
Comment (CharLotteiu) [reply sql-plan-management.md:28]:
```suggestion
-- Type one: Statements that use the `join` keyword but do not specify the associated columns with the `using` keyword to get the Cartesian product.
```
How about this?

File: sql-plan-management.md [hunk 9]
```diff
@@ -17,10 +17,73 @@ An SQL binding is the basis of SPM. The [Optimizer Hints](/optimizer-hints.md) d
...
+
+You can bypass syntax conflicts by using equivalent statements. For example, you can rewrite the above statements in the following ways:
+
+```sql
+-- First rewrite of type one statements: Add a `using` clause for the `join` keyword.
+create global binding for
+    select * from t t1 join t t2 using (a)
+using
+    select * from t t1 join t t2 using (a);
+-- Second rewrite of type one statements: Delete the `join` keyword.
```
[the remark this answers is not in the document]
Comment (CharLotteiu) [reply sql-plan-management.md:48]:
```suggestion

-- Second rewrite of type one statements: Delete the `join` keyword.
```

File: sql-plan-management.md [hunk 10]
```diff
@@ -17,10 +17,73 @@ An SQL binding is the basis of SPM. The [Optimizer Hints](/optimizer-hints.md) d
...
+create global binding for
+    select * from t t1, t t2
+using
+    select * from t t1, t t2;
+-- Rewrite of type two statements: Remove the `using` keyword from the `delete` statement.
```
[the remark this answers is not in the document]
Comment (CharLotteiu) [reply sql-plan-management.md:53]:
```suggestion

-- Rewrite of type two statements: Remove the `using` keyword from the `delete` statement.
```

File: sql-plan-management.md [hunk 11]
```diff
@@ -17,10 +17,73 @@ An SQL binding is the basis of SPM. The [Optimizer Hints](/optimizer-hints.md) d
...
+
+Here are two examples:
+
+```sql
+-- The hint takes effect in the following statement.
+create global binding for
+    insert into t1 select * from t2 where a > 1 and b = 1
+using
+    insert into t1 select /*+ use_index(@sel_1 t2, a) */ * from t2 where a > 1 and b = 1;
+-- The hint cannot take effect in the following statement.
```
[the remark this answers is not in the document]
Comment (CharLotteiu) [reply sql-plan-management.md:71]:
```suggestion

-- The hint cannot take effect in the following statement.
```

File: sql-plan-management.md [hunk 12]
```diff
@@ -132,7 +189,15 @@ To enable baseline capturing, set `tidb_capture_plan_baselines` to `on`. The def
...
-After automatic binding creation is enabled, the historical SQL statements in the Statement Summary are traversed every `bind-info-lease` (the default value is `3s`), and baseline capturing is automatically created for SQL statements that appear at least twice.
+After automatic binding creation is enabled, the historical SQL statements in the Statement Summary are traversed every `bind-info-lease` (the default value is `3s`), and a binding is automatically created for SQL statements that appear at least twice. For these SQL statements, TiDB automatically binds the execution plan recorded in Statement Summary.
+
+However, TiDB does not automatically capture bindings for the following types of SQL statements:
+
+- `EXPLAIN` and `EXPLAIN ANALYZE` statements.
+- SQL statements executed internally in TiDB, such as `SELECT` queries used for automatically loading statistical information.
+- SQL statements that are bound to a manually created execution plan.
+
+For `PREPARE` / `EXECUTE` statements and queries executed with binary protocols, TiDB automatically captures bindings for the real query statements, not for the `PREPARE` / `EXECUTE` statements.
```
[the remark this answers is not in the document]
Comment (CharLotteiu) [reply sql-plan-management.md:200]:
```suggestion
For `PREPARE` / `EXECUTE` statements and for queries executed with binary protocols, TiDB automatically captures bindings for the real query statements, not for the `PREPARE` / `EXECUTE` statements.
```

File: sql-plan-management.md [hunk 13]
```diff
@@ -48,42 +111,36 @@ explain select * from t1, t2 where t1.id = t2.id;
...
-select * from t where a >    1
--- parameterized:
-select * from t where a > ？
-```
-
 Each standardized SQL statement can have only one binding created using `CREATE BINDING` at a time. When multiple bindings are created for the same standardized SQL statement, the last created binding is retained, and all previous bindings (created and evolved) are marked as deleted. But session bindings and global bindings can coexist and are not affected by this logic.
 
 In addition, when you create a binding, TiDB requires that the session is in a database context, which means that a database is specified when the client is connected or `use ${database}` is executed.
 
 > **Note:**
```
[the remark this answers is not in the document]
Comment (CharLotteiu) [reply sql-plan-management.md:123]: It is suggested to be marked as a note in https://github.com/pingcap/docs/pull/1249#discussion_r296576774, but I agreed with you. I will update it later.

File: sql-plan-management.md (hunk 1 above)
Comment (eurekaka) [reply sql-plan-management.md:28]: 使用 join 但不没有 using，才会得到笛卡尔积，而不是说使用 using 关键字能得到笛卡尔积。

File: sql-plan-management.md [hunk 14]
```diff
@@ -48,42 +111,36 @@ explain select * from t1, t2 where t1.id = t2.id;
...
 >
-> The text must be the same before and after parameterization and hint removal for both the original SQL statement and the bound statement, or the binding will fail. Take the following examples:
+> The text must be the same before and after normalization and hint removal for both the original SQL statement and the bound statement, or the binding will fail. Take the following examples:
```
[the remark this answers is not in the document]
Comment (eurekaka) [reply sql-plan-management.md:125]: Yep.

**Pull request merged**